scripts/mirror-checker.py

```diff
@@ -1,77 +1,500 @@
+                status_text,
+                d,
+                output_dir / f"{cc}.txt",
+            )
+
+
+def check_all_mirrors(
+    n_jobs: int,
+    release: str,
+    show_non_mirrors: bool,
+    output_location: str,
+) -> int:
+    """Check the health of all country mirrors."""
+
+    print(datetime.now(timezone.utc))
+    print("Checking all mirrors...")
+    country_codes: list[str] = [c.alpha_2.lower() for c in countries]
+
+    print("Checking DNS information for mirrors...")
+    country_code_to_dns_data = get_dns_information(country_codes)
+
+    # Also collect IPs for the primary archive
+    primary_archive = {
+        "domain": "archive.ubuntu.com",
+        "A": get_arecords("archive.ubuntu.com"),
+        "CNAME": "",  # No CNAME for primary archive
+    }
+
+    # Print out all the mirror information found
+    _print_dns_information(
+        {
+            **{"primary": primary_archive},
+            **country_code_to_dns_data,
+        }
+    )
+
+    filtered_mirrors = filter_mirrors(country_code_to_dns_data, primary_archive)
+
+    mirror_aliases = get_multi_country_mirrors(country_code_to_dns_data)
+
+    _print_mirror_stats(
+        filtered_mirrors,
+        country_codes,
+        country_code_to_dns_data,
+        mirror_aliases,
+    )
+
+    # Get ready to check only active mirrors
+    mirror_status: dict[str, tuple[str, str, str | Path]] = {}
+
+    header = ["Mirror", "Status", "Domain", "Log"]
+
+    # Generate NA status result for non mirrors
+    for cc in filtered_mirrors["non_mirrors"]:
+        mirror_status[cc] = (
+            "NA",
+            "archive.ubuntu.com",
+            "-",
+        )
+
+    # Generate stub for mirror aliases
+    for mirror, aliases in mirror_aliases.items():
+        for cc in aliases:
+            mirror_status[cc] = (
+                "-",  # To be updated
+                f"{mirror}.archive.ubuntu.com",
+                "-",  # To be updated
+            )
+
+    # Remaining mirrors are real mirrors
+    mirrors_to_check = set()
+    for cc in country_codes:
+        if cc not in mirror_status:
+            domain = country_code_to_dns_data[cc]["CNAME"]
+            if domain == "":
+                domain = f"{cc}.archive.ubuntu.com"
+            mirrors_to_check.add((cc, domain))
+
+    # Also check the primary archive
+    mirrors_to_check.add(("primary", "archive.ubuntu.com"))
+
+    # Setup output directories
+    output_dir = Path(output_location)
+    output_dir.mkdir(exist_ok=True)
+
+    _multi_mirror_check(
+        mirrors_to_check,
+        release,
+        output_dir,
+        n_jobs,
+        mirror_status,
+    )
+
+    # Fill in results for aliases
+    for mirror, aliases in mirror_aliases.items():
+        for cc in aliases:
+            mirror_status[cc] = mirror_status[mirror]
+
+    table = PrettyTable(header)
+    table.add_row(["primary", *mirror_status.pop("primary")])
+    for cc, status in sorted(mirror_status.items()):
```

Comment:
suggestion: it's great news that all the mirrors look healthy right now, but we expect later that won't be the case.  What do you think about sorting on Status, with `!ok` at the bottom?  It would be nice to have failing mirrors on the bottom, which means we can just check the last lines of the report for failing mirrors and if that isn't the case, we don't need to read the log much further.  WDYT?

Reply:
Yeah that sounds much better to me.